### PR TITLE
fix: file_replace avoid identation, find and replace actual content

### PR DESCRIPTION
**Pull request by ArslanSaleem**
> [!IMPORTANT]
> Introduces indentation-insensitive file content replacement utility and updates existing logic to use it, with comprehensive tests.
> 
>   - **Behavior**:
>     - Replaces string replacement logic in `file_str_replace()` in `file_ops.py` with `file_replace_utils.file_replace()` to handle indentation-insensitive replacements.
>     - Updates `get_file_from_conversation_messages()` in `files.py` to use `file_replace()` for content replacement.
>   - **Utilities**:
>     - Adds `file_replace.py` with functions for indentation-insensitive content replacement, including `file_replace()`, `normalize_indentation()`, `block_to_indentation_flexible_pattern()`, `reindent_like()`, and `indent_insensitive_contains()`.
>   - **Tests**:
>     - Adds `test_file_replace.py` with tests for `file_replace()` and its helper functions, covering various XML replacement scenarios and edge cases.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=sinaptik-ai%2Fpanda-agi&utm_source=github&utm_medium=referral)<sup> for c49a05b497029d049fec674d5ccccefe3f7cebcf. You can [customize](https://app.ellipsis.dev/sinaptik-ai/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: examples/ui/backend/panda_agi/utils/file_replace.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    pat = block_to_indentation_flexible_pattern(search_block)
+    return re.search(pat, content, flags=re.DOTALL) is not None
+
+
+def file_replace(content: str, search_block: str, replace_block: str, count: int = 0):
+    """
+    Replace occurrences of search_block ignoring indentation differences only.
+    Returns (new_content, num_replacements).
+    """
+    pat = block_to_indentation_flexible_pattern(search_block)
```
[the remark this answers is not in the document]
Comment (ArslanSaleem) [reply examples/ui/backend/panda_agi/utils/file_replace.py:58]: add check for empty search_block
